Merge Conflict/Assets/Scripts/Menu/UpgradeMenu.cs

```diff
@@ -0,0 +1,115 @@
+{
+    [Header("Buttons")]
+    public Button mergeXPButton;
+    public Button moneyWhenTrashedButton;
+    public Button spawnChanceWhenTrashDiscardedButton;
+    public Button spawnIntervalButton;
+
+    [Header("Button Texts")]
+    public TextMeshProUGUI mergeXPButtonText;
+    public TextMeshProUGUI moneyWhenTrashedButtonText;
+    public TextMeshProUGUI spawnChanceWhenTrashDiscardedButtonText;
+    public TextMeshProUGUI spawnIntervalButtonText;
+
+    [Header("Description Texts")]
+    public TextMeshProUGUI mergeXPDescriptionText;
+    public TextMeshProUGUI moneyWhenTrashedDescriptionText;
+    public TextMeshProUGUI spawnChanceWhenTrashDiscardedDescriptionText;
+    public TextMeshProUGUI spawnIntervalDescriptionText;
+
+    [Header("UI Texts")]
+    public TextMeshProUGUI levelText;
+    public TextMeshProUGUI moneyText;
+
+
+
+    void Start()
+    {
+        mergeXPButton.onClick.AddListener(() => UpgradeMergeXP(Upgrades.MergeXPUpgrade));
+        moneyWhenTrashedButton.onClick.AddListener(() => UpgradeMoneyWhenTrashed(Upgrades.MoneyWhenTrashedUpgrade));
+        spawnChanceWhenTrashDiscardedButton.onClick.AddListener(() => UpgradeSpawnChanceWhenTrashDiscarded(Upgrades.SpawnChanceWhenTrashDiscardedUpgrade));
+        spawnIntervalButton.onClick.AddListener(() => UpgradeSpawnInterval(Upgrades.SpawnIntervalUpgrade));
+
+        // TODO: delete this test money function later
+#if UNITY_EDITOR
+        MoneyHandler.Instance.AddMoney(100000);
+#endif
+
+        UpdateUI();
+    }
+
+    void Update()
+    {
+    }
+
+    public void UpdateUI()
+    {
+        levelText.text = "Level " + ExperienceSystem.ExperienceHandler.GetCurrentLevel().ToString();
+        moneyText.text = MoneyHandler.Instance.Money.ToString() + " $";
+
+        UpdateDescriptionsTexts();
+        UpdateButtons();
+    }
+
+
+    void UpdateButtons()
+    {
+        // Check if next level is available
+        mergeXPButtonText.text = Upgrades.MergeXPUpgrade.isAtMaxLevel() ? "Max" : $"{Upgrades.MergeXPUpgrade.CostForNextLevel[Upgrades.MergeXPUpgrade.Level]} $";
+        moneyWhenTrashedButtonText.text = Upgrades.MoneyWhenTrashedUpgrade.isAtMaxLevel() ? "Max" : $"{Upgrades.MoneyWhenTrashedUpgrade.CostForNextLevel[Upgrades.MoneyWhenTrashedUpgrade.Level]} $";
+        spawnChanceWhenTrashDiscardedButtonText.text = Upgrades.SpawnChanceWhenTrashDiscardedUpgrade.isAtMaxLevel() ? "Max" : $" {Upgrades.SpawnChanceWhenTrashDiscardedUpgrade.CostForNextLevel[Upgrades.SpawnChanceWhenTrashDiscardedUpgrade.Level]} $";
+        spawnIntervalButtonText.text = Upgrades.SpawnIntervalUpgrade.isAtMaxLevel() ? "Max" : $"{Upgrades.SpawnIntervalUpgrade.CostForNextLevel[Upgrades.SpawnIntervalUpgrade.Level]} $";
+    }
+
+    void UpdateDescriptionsTexts()
+    {
+        mergeXPDescriptionText.text = $"{Upgrades.MergeXPUpgrade.GetCurrentLevelDescription()}" + (!Upgrades.MergeXPUpgrade.isAtMaxLevel() ? $"\n -> Next Level ({Upgrades.MergeXPUpgrade.GetNextLevelDescription()})" : "");
```

Comment:
this is fine, but cumbersome to change. SInce the logic is the same for all upgrades, i would simply create a function for this, that takes in the upgrade and returns the DescriptionText.

Reply:
done